File: backend/api/src/main/scala/com/github/ai/split/api/response/PutExpenseResponse.scala
package com.github.ai.split.api.response

import com.github.ai.split.api.ExpenseDto
import zio.json.{DeriveJsonEncoder, JsonEncoder}

case class PutExpenseResponse(
  expense: ExpenseDto
)

object PutExpenseResponse {
  implicit val encoder: JsonEncoder[PutExpenseResponse] = DeriveJsonEncoder.gen[PutExpenseResponse]
}

File: backend/api/src/main/java/com/github/ai/split/api/response/PutExpenseResponse.java
package com.github.ai.split.api.response;

import com.github.ai.split.api.ExpenseDto;
public class PutExpenseResponse {

    public ExpenseDto expense = new ExpenseDto();

    public PutExpenseResponse() {}

    public PutExpenseResponse(ExpenseDto expense) {
        this.expense = expense != null ? expense : new ExpenseDto();
    }
}